src/main/java/sparksoniq/jsoniq/runtime/iterator/DataFrameRuntimeIterator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Stefan Irimescu, Can Berker Cikis
+ *
+ */
+
+package sparksoniq.jsoniq.runtime.iterator;
+
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+
+import java.util.List;
+
+public abstract class DataFrameRuntimeIterator extends RDDRuntimeIterator {
```

Comment:
Class defined for DataFrame only Iterators. eg. Parquet file and Structured Json File.

In the old implementation this would have a method isDataFrame which would always return true.

New implementation uses this class while checking if a function's iterator is a subclass of this.